### DISCUSSION TITLE
Glue together tiny road/footway section

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#2436] [Ideas]
In some places, especially well mapped ones, roads are sometimes split into tiny pieces.

https://www.openstreetmap.org/way/218427503#map=19/50.08812/19.89779 is not unusual

Multiple people (one more right now) commented to me that this segmentation makes SC editing a bit irritating in some places.

Maybe it would be feasible to present together such tiny pieces during quest generation as one quest. Ways would not be actually merged in OSM, just solving quest would apply result to all of them.

Main complexity is for splitting - it should be possible to make splits possible without them getting merged immediately again (and maybe encourage a tiny bit splitting where OSM ways are already separate),

**Comment by HolgerJeromin** [comment]
But perhaps this split was done because the section is different.
How do you want to decide?
Which quest should merge quests?

**Reply by matkoniecz** [reply]
It would not merge ways in OSM. It would present multiple tiny ways for solution at once and apply answer to each way - without merging it in OSM

I edited my text a bit to clarify it a bit and write what I meant.

**Comment by smichel17** [comment]
> Main complexity is for splitting - it should be possible to make splits possible without them getting merged immediately again

Selecting "differs along the way" should automatically "split" the way back to the small segments that are currently mapped, rather than prompting the user to split it in a new location. So there needs to be a way (ha!) to exclude certain ways from this merging feature.

**Comment by smichel17** [comment]
Also, I'll note that this feature is a special case of https://github.com/streetcomplete/StreetComplete/issues/826#issuecomment-362786776, https://github.com/streetcomplete/StreetComplete/issues/2324, etc
